### ISSUE TITLE
Add Colourspace Conversion / Manipulation Module to the Standard Library

### ISSUE THREAD

**Issue by stellarpower** [nushell/nushell#10599]
Hi,

I'm looking to do some basic colourspace work - 

- parse some hex strings
- extract the hue, discard the rest
- set the saturation and value
- convert back to RGB
- etc.

I think it'd be nice if this were available in the standard library, so I can do as a quick shell script.

Cheers

**Comment by Harsh1s** [comment]
> Hi,
> 
> I'm looking to do some basic colourspace work -
> 
> * parse some hex strings
> * extract the hue, discard the rest
> * set the saturation and value
> * convert back to RGB
> * etc.
> 
> I think it'd be nice if this were available in the standard library, so I can do as a quick shell script.
> 
> Cheers

Hello there, Harsh this side, I am kind of new to rust but have done some small personal projects with it. Nevertheless, I am really interested in this project. Since this is supposed to be a good first issue, I wanted to tackle this one. 
Forgive me if I sound stupid but could you clarify the issue a bit more? I get the part that a hex string is to be parsed and I need to extract the RGB value from that. I am aware of the fact that I'll need to convert them to another color space like HSL to extract the hue, and then saturation and value can be set before changing it back to RGB. 
I would appreciate it if you could provide an example of sorts, like what kind of string do we have to parse. I am grateful for your patience.

**Comment by amtoine** [comment]
@Harsh1s :wave: :yum: 

thanks for considering helping use make Nushell better :pray: 

just wanted to bring a little precision on this issue, @stellarpower are talking about the standard library of Nushell, which is written in Nushell, not Rust :wink:

**Comment by Harsh1s** [comment]
> @Harsh1s 👋 😋
> 
> thanks for considering helping use make Nushell better 🙏
> 
> just wanted to bring a little precision on this issue, @stellarpower are talking about the standard library of Nushell, which is written in Nushell, not Rust 😉

Oh, I feel even more stupid now. Anyways, thanks for your time.

**Comment by stellarpower** [comment]
I am, but, maybe wrapping a rust library is the easiest way to achieve this(?). In terms of not solving a problem already solved, and writing it all out in nu source code. Even if that's in a sense more desireable than binding a crate. I don't think it was stupid BTW.

@Harsh1s - [this](https://github.com/ndudnicz/selenite) is what I ended up using instead, so, illustrates some basics of what I am thinking of.

**Comment by Harsh1s** [comment]
> I am, but, maybe wrapping a rust library is the easiest way to achieve this(?). In terms of not solving a problem already solved, and writing it all out in nu source code. Even if that's in a sense more desireable than binding a crate. I don't think it was stupid BTW.
> 
> @Harsh1s - [this](https://github.com/ndudnicz/selenite) is what I ended up using instead, so, illustrates some basics of what I am thinking of.

I totally agree with you on this one. Even I'm inclined to add this functionality in Rust itself as it would make sense to make a new package for managing all color related operations in Nu if required.

**Comment by LoicRiegel** [comment]
Hi, if you're still interested in this, I think this would be a perfect candidate for a new plugin.
I didn't see any color manipulation plugin in [awesome-nu](https://github.com/nushell/awesome-nu), so I guess no one has done it yet.

And I also agree that wrapping a rust library would save you a lot of work
